var/spack/repos/builtin/packages/perl-dbd-mysql/package.py

```diff
@@ -22,6 +33,18 @@ class PerlDbdMysql(PerlPackage):
     version("4.043", sha256="629f865e8317f52602b2f2efd2b688002903d2e4bbcba5427cb6188b043d6f99")
 
     depends_on("perl-devel-checklib", type="build", when="@4.050:")
-    depends_on("perl-test-deep", type=("build", "run"))
-    depends_on("perl-dbi", type=("build", "run"))
-    depends_on("mysql-client")
+
+    with default_args(type=("build", "link", "run")):
+        # Does it's own version check and mariadb doesn't conform to it's
+        # strict checking. This could probably be patched in the future.
+        depends_on("mysql@4:", when="@4")
+        depends_on("mysql@8", when="@5")
```

Comment:
```suggestion
        depends_on("mysql@8:", when="@5")
```
Possible typo? Apologies if this constraint was intentional.

Reply:
Sadly, no. It's looking for exactly mysql 8. The source of many downstream woes.